### DISCUSSION TITLE
vscode launch.json example

### DISCUSSION THREAD

**Discussion by parnic** [henrygd/beszel#146] [General]
Since the .vscode folder is gitignored, I assume that means there's no desire to keep a launch.json file in the repo for ease of debugging, so I'll post my working config here for others:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Start hub server",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "cwd": "${workspaceFolder}/beszel",
            "program": "${workspaceFolder}/beszel/cmd/hub/hub.go",
            "args": ["serve"]
        }
    ]
}
```

Change the "serve" arg to another supported mode if you want to debug that mode, e.g. `admin`, `migrate`, `update`.

Note that you'll have to build the frontend files first [as explained in the readme](https://github.com/henrygd/beszel?tab=readme-ov-file#hub-2), but once you've done that, you should be able to debug the application with this.

**Comment by henrygd** [comment]
Nice, thanks! I'll try it out.
